dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt

```diff
@@ -1,33 +1,153 @@
+                "XXXsomeUUIDXXX"
+            )
+        )
+        val dataId = dataManager.addDataSet(storableDataSet, correlationId)
+        `when`(edcClientMock.selectDataById(dataId, correlationId)).thenThrow(ServerException::class.java)
+        assertThrows<ServerException> {
+            dataManager.getDataSet(dataId, DataType(storableDataSet.dataType.name), correlationId)
         }
     }
+
+    @Test
+    fun `check that an exception is thrown when non matching dataId to dataType pair is requested from data storage`() {
+        val storableDataSet = addCompanyAndReturnStorableEuTaxonomyDataSetForNonFinacialsForIt()
+        val storableDataSetAsString = objectMapper.writeValueAsString(storableDataSet)
+        `when`(edcClientMock.insertData(correlationId, storableDataSetAsString)).thenReturn(
+            InsertDataResponse(
+                "XXXsomeUUIDXXX"
+            )
+        )
+        val dataId = dataManager.addDataSet(storableDataSet, correlationId)
+        val thrown = assertThrows<InvalidInputApiException> {
+            dataManager.getDataSet(dataId, DataType("eutaxonomy-financials"), correlationId)
+        }
+        assertEquals(
+            "The data with the id: XXXsomeUUIDXXX is registered as type eutaxonomy-non-financials by " +
+                "Dataland instead of your requested type eutaxonomy-financials.",
+            thrown.message
+        )
+    }
+
+    @Test
+    fun `check that an exception is thrown if the received data from the data storage is empty`() {
+        val storableDataSet = addCompanyAndReturnStorableEuTaxonomyDataSetForNonFinacialsForIt()
+        val storableDataSetAsString = objectMapper.writeValueAsString(storableDataSet)
+        `when`(edcClientMock.insertData(correlationId, storableDataSetAsString)).thenReturn(
+            InsertDataResponse(
+                "XXXsomeUUIDXXX"
+            )
+        )
+        val dataId = dataManager.addDataSet(storableDataSet, correlationId)
+        `when`(edcClientMock.selectDataById(dataId, correlationId)).thenReturn("")
+        val thrown = assertThrows<ResourceNotFoundApiException> {
+            dataManager.getDataSet(dataId, DataType("eutaxonomy-non-financials"), correlationId)
+        }
+        assertEquals("No dataset with the id: XXXsomeUUIDXXX could be found in the data store.", thrown.message)
+    }
+
+    @Test
+    fun `check that an exception is thrown if the received data from the data storage has an unexpected type`() {
+        val storableDataSet = addCompanyAndReturnStorableEuTaxonomyDataSetForNonFinacialsForIt()
+        val storableDataSetAsString = objectMapper.writeValueAsString(storableDataSet)
+        `when`(edcClientMock.insertData(correlationId, storableDataSetAsString)).thenReturn(
+            InsertDataResponse(
+                "XXXsomeUUIDXXX"
+            )
+        )
+        val dataId = dataManager.addDataSet(storableDataSet, correlationId)
+        val unexpectedDataTypeName = "eutaxonomy-financials"
```

Comment:
Die Variable kommt nur einmal vor warum nicht direkt in die Funktion

Reply:
inlined